tests/unit-test.el

```diff
@@ -9,3 +10,81 @@
   (it "returns the ns name of its argument"
     (expect (cljr--ns-name "com.corp.foo") :to-equal "foo")
     (expect (cljr--ns-name "foo") :to-equal "foo")))
+
+(defmacro cljr--with-clojure-temp-file (filename &rest body)
+  (declare (indent 1))
+  `(with-temp-buffer
+     (setq-local buffer-file-name ,filename)
+     (delay-mode-hooks (clojure-mode))
+     ,@body))
+
+(describe "cljr--insert-require-libspec"
+  :var (after-inserting-a
+        after-inserting-b)
+  (before-each
+    (cljr--with-clojure-temp-file "libspec.clj"
+      (insert "(ns lib)\n")
+      (cljr--insert-require-libspec "[com.a :as a]")
+      (setq after-inserting-a (buffer-string))
+      (cljr--insert-require-libspec "[com.b :as b]")
+      (setq after-inserting-b (buffer-string))))
+
+  (it "adds a new require over an empty ns form"
+    (expect after-inserting-a :to-equal
+            "(ns lib\n  (:require [com.a :as a]))\n"))
+  (it "adds a new require over a ns with existing requires"
+    (expect after-inserting-b :to-equal
+            "(ns lib
+  (:require [com.a :as a]
+            [com.b :as b]))
+")))
+
+(describe "cljr--ns-alias-at-point"
+  (it "returns the short alias before the /"
+    (expect (with-temp-buffer
+              (insert "aba/")
+              (cljr--ns-alias-at-point))
+            :to-equal "aba"))
+
+  (it "removes namespace keyword ::"
+    (expect (with-temp-buffer
+              (insert "::ns-name/")
+              (cljr--ns-alias-at-point))
+            :to-equal "ns-name"))
+
+  (it "removes deref operator"
```

Comment:
Couple more things that shoud be likely removed:

```
`
#'
```

Reply:
Oh, I think that's new behavior. I'm happy to add more cases, but was trying to keep this focused on the current behavior.

I also noticed it didn't seem to like dotted namespace aliases like `clojure.set/` as it just returned "set". So I think there are a couple of cases we should account for. My preference would be to make that a different issue, which I'm also happy to tackle but trying to keep new behavior separate. What is your preference here?

Reply:
Alright, let's create an issue and briefly link to it in a fixme

Reply:
https://github.com/clojure-emacs/clj-refactor.el/issues/524

Reply:
Added fixme + pending tests with https://github.com/clojure-emacs/clj-refactor.el/pull/522/commits/ead4813737ea4e21d5aa1c8422a1ba247321b481